### DISCUSSION TITLE
Service Client and Early Bound generation

### DISCUSSION THREAD

**Discussion by Kevmscotland** [microsoft/PowerPlatform-DataverseServiceClient#160] [Q&A] [answered]
Previously I've used the CrmSvcUtil to generate EarlyBound classes which creates the OrganisationServiceContext (which requires the old IOrganisationService).
This enables me to keep everything strongly typed and use Linq queries for data retrieval etc.

Is there a proposed way or guide on how to go about generating Earlybound classes to create a Context for using the new Dataverse Service Client?

**Comment by MattB-msft** [comment, marked as answer]
it works exactly the same. 
Just pass the Dataverse ServiceClient to the OrganizationServiceContext.

**Reply by Kevmscotland** [reply]
Doh. I hadn't even tried. I thought I saw it implemented a different interface at a glance and assumed it wouldn't work! Im a fool. Thanks!

**Reply by MattB-msft** [reply]
Dataverse Service Client is an IOrganizaitonService :)

**Reply by sabrish** [reply]
@MattB-msft - are the early bound classes generated dot net core compatible? It wasn't when I checked a long time ago.

Thanks

**Reply by BetimBeja** [reply]
Isn't it easier trying than asking... you lose 10 minutes to try... and make a lot people lose time following this type of question 😞

**Reply by Kevmscotland** [reply]
Its a good discussion point. Now other people who may not have known now have the answer thanks to Matt. 😄

**Reply by MattB-msft** [reply]
Yes code generated by crmsvcutil is .net core compatible.

We also released and update to the generator to try to help prevent folks from running into issues with messages that use unreleased types.

**Reply by xairrick** [reply]
I normally use the generator that is in the XrmToolbox (which I believe uses the CrmSrvUtil) and have found the early bound classes & context work without any modifications in .NET Core 3.1.    The only thing that worries me is that the OrganizationServiceContext  is using IOrganizationService  and not IOrganizationServiceAsync or IOrganizationServiceAsync2

**Reply by jordimontana82** [reply]
@sabrish, yeah, they were .net core compatible since first time I tried back in Feb 2020 :)
